scripts/nlp_language_modeling/prepare_packed_ft_dataset.py

```diff
@@ -83,12 +83,26 @@ def tokenize_dataset(cfg: 'DictConfig'):
     # using the same template as SFT/PEFT script. This may be overkill but guarantees the preprocess settings
     # are identical to normal SFT training
     data_cfg = cfg.model.data.train_ds
+    pad_seq_length_to_mult=16
+    cp_size = cfg.model.context_parallel_size
+
+    # if context parallel is used, each individual data length in one packed dataset sample
+    # needs to be a multiple of (cp_size * 2): https://github.com/NVIDIA/TransformerEngine/pull/641
+    if cp_size > 1:
+      # Function to calculate the LCM of two numbers
+      def lcm(x, y):
+        larger = max(x, y)
+        for i in range(larger, x * y + 1):
+          if i % x == 0 and i % y == 0:
+              return i
+      pad_seq_length_to_mult = lcm(pad_seq_length_to_mult, cp_size * 2)
```

Comment:
Do we need such a complex function? I guess `max(pad_seq_length_to_mult, cp_size*2)` is enough?

Reply:
changed to `max(pad_seq_length_to_mult, cp_size*2)` in the latest commit